src/main/PiMost.ts

```diff
@@ -135,26 +112,23 @@ export class PiMost {
         }
       })
 
-      this.socketMostClient.on(
-        Os8104Events.SocketMostMessageRxEvent,
-        (message: RawMostRxMessage) => {
-          const type = fBlocks[message.fBlockID]
-          if (message.opType === 15) {
-            console.log('most error', message)
-          }
-          if (type === this.timeoutType) {
-            this.subscriptionTimer!.refresh()
-          }
-          this.interfaces?.[type]?.parseMessage(message)
+      this.socketMostClient.on(Os8104Events.SocketMostMessageRxEvent, (message: MostRxMessage) => {
+        const type = fBlocks[message.fBlockID]
+        if (message.opType === 15) {
+          console.log('most error', message)
+        }
+        if (type === this.timeoutType) {
+          this.subscriptionTimer!.refresh()
         }
-      )
+        this.interfaces?.[type]?.parseMessage(message)
+      })
     })
   }
   stream(stream: Stream) {
     this.socketMostClient.stream(stream)
   }
 
-  sendMessage(message: SocketMostSendMessage) {
+  sendMessage = (message: SocketMostSendMessage) => {
```

Comment:
Initialising this way binds it to this from the get go

Reply:
ah, amazing, thanks Yoni!